etcdserver/api/v3rpc/lease.go

```diff
@@ -86,7 +86,7 @@ func (ls *LeaseServer) LeaseKeepAlive(stream pb.Lease_LeaseKeepAliveServer) erro
 		}
 
 		if err != nil {
-			return err
+			return handleLeaseEOF(err)
```

Comment:
keepAlive uses leaseRenew. i think we might want to catch EOF at that function.
https://github.com/coreos/etcd/blob/master/etcdserver/api/v3rpc/lease.go#L82